### ISSUE TITLE
collections.Sort looses slice type info (which makes it fail in .Paginate)

### ISSUE THREAD

**Issue by gbmhunter** [gohugoio/hugo#6023]
I use the paginator in the following fashion:

```
{{ range (.Paginate (first 20 ( sort (where .Site.RegularPages "Section" "posts") "Date" "desc" )) 5 ).Pages }}
   <do stuff here>
{{ end }}
```

This worked fine in `v0.54`. However, upon upgrading to `v0.56.6` (this includes the paginator bug fix), I get the following error when trying to either run the server or perform a build:

```
Error: Error building site: failed to render pages: render of "home" failed: "C:\Users\gbmhu\code\blog\layouts\index.html:11:14": execute of template failed: template: index.html:11:14: executing "main" at <.Paginate>: error calling Paginate: cannot convert type []page.Page to Pages
```

I am running `hugo` on Windows 10, installed by chocolaty. The repo that contains the code which causes the crash can be found at https://github.com/gbmhunter/blog (trying to run that with `v0.55.6` should cause the exact same error above).

I even tried simplifying the paginator to what is specified in the tutorial, i.e.:

```
{{ $paginator := .Paginate (where .Pages "Type" "posts") }}
{{ template "_internal/pagination.html" . }}
{{ range $paginator.Pages }}
   {{ .Title }}
{{ end }}
```

The crash still occurred.

**Comment by bep** [comment]
I have not tested this, but you could try this for a workaround:

```bash
{{ $pages := (first 20 ( sort (where .Site.RegularPages "Section" "posts") "Date" "desc" ) | slice }}
{{ range (.Paginate $pages) 5 ).Pages }}
```

**Comment by gbmhunter** [comment]
Thanks for the investigation and fix for v0.56!
